packages/api/tests/lambdas/sf-event-sqs-to-db-records/test-write-execution.js

```diff
@@ -63,119 +57,417 @@ test.beforeEach((t) => {
   t.context.executionName = cryptoRandomString({ length: 5 });
   t.context.executionArn = `arn:aws:states:us-east-1:12345:execution:${stateMachineName}:${t.context.executionName}`;
 
+  t.context.workflowStartTime = Date.now();
+  t.context.workflowTasks = {
+    task1: {
+      key: 'value',
+    },
+  };
+  t.context.workflowName = 'TestWorkflow';
+
   t.context.cumulusMessage = {
     cumulus_meta: {
-      workflow_start_time: 122,
+      workflow_start_time: t.context.workflowStartTime,
       cumulus_version: t.context.postRDSDeploymentVersion,
       state_machine: t.context.stateMachineArn,
       execution_name: t.context.executionName,
     },
     meta: {
       status: 'running',
+      workflow_name: t.context.workflowName,
+      workflow_tasks: t.context.workflowTasks,
+    },
+    payload: {
+      foo: 'bar',
     },
   };
-
-  t.context.hasAsyncOpStub = hasAsyncOpStub;
-  t.context.hasAsyncOpStub.resetHistory();
-
-  t.context.hasParentExecutionStub = hasParentExecutionStub;
-  t.context.hasParentExecutionStub.resetHistory();
 });
 
 test.after.always(async (t) => {
   const {
     executionModel,
   } = t.context;
   await executionModel.deleteTable();
-  sandbox.restore();
   await t.context.knex.destroy();
   await t.context.knexAdmin.raw(`drop database if exists "${t.context.testDbName}"`);
   await t.context.knexAdmin.destroy();
 });
 
-test('shouldWriteExecutionToRDS returns false for pre-RDS deployment execution message', async (t) => {
-  const { cumulusMessage, knex, preRDSDeploymentVersion } = t.context;
-  t.false(await shouldWriteExecutionToRDS(
-    {
+test('shouldWriteExecutionToRDS returns false for pre-RDS deployment execution message', (t) => {
+  const { cumulusMessage, preRDSDeploymentVersion } = t.context;
+  t.false(shouldWriteExecutionToRDS({
+    cumulusMessage: {
       ...cumulusMessage,
       cumulus_meta: {
         ...cumulusMessage.cumulus_meta,
         cumulus_version: preRDSDeploymentVersion,
       },
     },
-    1,
-    knex
-  ));
+  }));
 });
 
-test.serial('shouldWriteExecutionToRDS returns true for post-RDS deployment execution message if all referenced objects exist', async (t) => {
+test('shouldWriteExecutionToRDS returns true for post-RDS deployment execution message if all referenced objects exist', (t) => {
   const {
-    knex,
     cumulusMessage,
   } = t.context;
 
-  t.context.hasAsyncOpStub.withArgs(cumulusMessage).resolves(true);
-  t.context.hasParentExecutionStub.withArgs(cumulusMessage).resolves(true);
-
   t.true(
-    await shouldWriteExecutionToRDS(
+    shouldWriteExecutionToRDS({
       cumulusMessage,
-      1,
-      knex
-    )
+      collectionCumulusId: 1,
+      asyncOperationCumulusId: 2,
+      parentExecutionCumulusId: 3,
+    })
   );
 });
 
-test.serial('shouldWriteExecutionToRDS returns false if error is thrown', async (t) => {
+test('shouldWriteExecutionToRDS returns false if any referenced objects are missing', async (t) => {
```

Comment:
See above concern re: write logic - we need changes/tests around async/collection being not present in the message.

Reply:
With the factor, see the new tests:

https://github.com/nasa/cumulus/pull/2005/files?file-filters%5B%5D=.js#diff-74fdb6cc3f08eb88163d987562f8fd73a7181eba22a2b71c6b2490d5d10884a8R99
https://github.com/nasa/cumulus/pull/2005/files?file-filters%5B%5D=.js#diff-74fdb6cc3f08eb88163d987562f8fd73a7181eba22a2b71c6b2490d5d10884a8R112
https://github.com/nasa/cumulus/pull/2005/files?file-filters%5B%5D=.js#diff-74fdb6cc3f08eb88163d987562f8fd73a7181eba22a2b71c6b2490d5d10884a8R125